shared/src/types/token.rs

```diff
@@ -44,15 +45,16 @@ impl Amount {
         self.micro as Change
     }
 
-    /// Spend a given amount
+    /// Spend a given amount.
     /// Panics when given `amount` > `self.micro` amount.
     pub fn spend(&mut self, amount: &Amount) {
-        self.micro -= amount.micro
+        self.micro = self.micro.checked_sub(amount.micro).unwrap();
```

Comment:
This and `receive` were in fact not panicking, but instead the values were under-flowing/overflowing. Updated to cause a panic instead